src/answers-headless.ts

```diff
@@ -268,14 +276,41 @@ export default class AnswersHeadless {
     return results;
   }
 
-  executeFilterSearch(sectioned: boolean, fields: SearchParameterField[]): Promise<FilterSearchResponse> {
-    return this.core.filterSearch({
-      input: this.state.query.query || '',
+  async executeFilterSearch(
+    filterSearchId: string,
+    sectioned: boolean,
+    fields: SearchParameterField[]
+  ): Promise<FilterSearchResponse | undefined> {
+    const thisRequestId = this.httpManager.updateRequestId('filterSearch');
+    const verticalKey = this.state.vertical.key;
+    if (!verticalKey) {
+      console.error('no verticalKey supplied for filter search');
+      return;
+    }
+    if (!filterSearchId || !this.state.filterSearch[filterSearchId]) {
+      console.error(`invalid filterSearch id: "${filterSearchId}""`);
```

Comment:
Can this error explain how the id is invalid?

Reply:
updated, should be clearer now